internal/pkg/admission/reconciler.go

```diff
@@ -38,41 +36,35 @@ func (errorList errorList) Error() string {
 
 func (r *Reconciler) ReconcileDeletion(
 	ctx context.Context,
-	clusterAdmissionPolicy *policiesv1alpha2.ClusterAdmissionPolicy,
+	policyServer *policiesv1alpha2.PolicyServer,
 ) error {
 	errors := errorList{}
-	r.Log.Info("Removing deleted policy from PolicyServer ConfigMap")
-	if err := r.reconcilePolicyServerConfigMap(ctx, clusterAdmissionPolicy, RemovePolicy); err != nil {
-		r.Log.Error(err, "ReconcileDeletion: cannot update ConfigMap")
-		errors = append(errors, err)
-	}
 
-	r.Log.Info("Removing ValidatingWebhookConfiguration associated with deleted policy")
-	validatingWebhookConf := &admissionregistrationv1.ValidatingWebhookConfiguration{
+	deployment := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: clusterAdmissionPolicy.Name,
+			Name:      policyServer.NameWithPrefix(),
+			Namespace: r.DeploymentsNamespace,
 		},
 	}
-	if err := r.Client.Delete(ctx, validatingWebhookConf); err != nil && !apierrors.IsNotFound(err) {
-		r.Log.Error(err, "ReconcileDeletion: cannot delete ValidatingWebhookConfiguration")
+	err := r.Client.Delete(ctx, deployment)
+	if err != nil {
+		r.Log.Error(err, "ReconcileDeletion: cannot delete PolicyServer Deployment %s", policyServer.Name)
 		errors = append(errors, err)
 	}
 
-	r.Log.Info("Removing MutatingWebhookConfiguration associated with deleted policy")
-	mutatingWebhookConf := &admissionregistrationv1.MutatingWebhookConfiguration{
+	certificateSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: clusterAdmissionPolicy.Name,
+			Name:      policyServer.NameWithPrefix(),
+			Namespace: r.DeploymentsNamespace,
 		},
 	}
-	if err := r.Client.Delete(ctx, mutatingWebhookConf); err != nil && !apierrors.IsNotFound(err) {
-		r.Log.Error(err, "ReconcileDeletion: cannot delete MutatingWebhookConfiguration")
-		errors = append(errors, err)
-	}
 
-	if err := r.reconcilePolicyServerDeployment(ctx); err != nil {
-		r.Log.Error(err, "ReconcileDeletion: cannot reconcile PolicyServer deployment")
+	err = r.Client.Delete(ctx, certificateSecret)
+	if err != nil {
+		r.Log.Error(err, "ReconcileDeletion: cannot delete PolicyServer Certificate Secret %s", policyServer.Name)
 		errors = append(errors, err)
 	}
+
```

Comment:
nitpicking: add a `// TODO: delete X, Y, Z` to track the other resources that have to be deleted

Reply:
good point! I just added it, so we don't forget. See last commit